lucene/core/src/java/org/apache/lucene/index/PostingsEnum.java

```diff
@@ -67,6 +67,56 @@ public static boolean featureRequested(int flags, short feature) {
   protected PostingsEnum() {
   }
 
+  /**
+   * Dummy PostingsEnum object.
+   */
+  public static PostingsEnum DUMMY_INSTANCE = new PostingsEnum() {
```

Comment:
Please look at org.apache.lucene.search.DocIdSetIterator.empty() 
There's subtle exhausting logic which needs to be supported and you can follow naming as well.